CI failing due to hot plug CPUs

Summarizing 4 Failures:

[Fail] Hot plug CPUs container with CPU period and quota [It] should have 3 CPUs 
/home/jenkins/jenkins_slave/workspace/kata-containers-tests-ubuntu-17-10-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:91

[Fail] Hot plug CPUs container with CPU period and quota [It] should have 4 CPUs 
/home/jenkins/jenkins_slave/workspace/kata-containers-tests-ubuntu-17-10-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:91

[Fail] Hot plug CPUs container with CPU constraint [It] should have 3 CPUs 
/home/jenkins/jenkins_slave/workspace/kata-containers-tests-ubuntu-17-10-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:108

[Fail] Hot plug CPUs container with CPU constraint [It] should have 3 CPUs 
/home/jenkins/jenkins_slave/workspace/kata-containers-tests-ubuntu-17-10-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:108

Ran 98 of 159 Specs in 543.348 seconds
FAIL! -- 94 Passed | 4 Failed | 0 Pending | 61 Skipped --- FAIL: TestIntegration (567.07s)
FAIL

--- a/integration/docker/cpu_test.go
+++ b/integration/docker/cpu_test.go
@@ -79,7 +79,6 @@ var _ = Describe("Hot plug CPUs", func() {
 
 	DescribeTable("container with CPU period and quota",
 		func(quota, period int, fail bool) {
-			Skip("Issue: https://github.com/kata-containers/tests/issues/250")
 			args = append(args, "--cpu-quota", fmt.Sprintf("%d", quota),
 				"--cpu-period", fmt.Sprintf("%d", period), Image, "sh", "-c", "sleep 5; nproc")
 			vCPUs = (quota + period - 1) / period
@@ -99,7 +98,6 @@ var _ = Describe("Hot plug CPUs", func() {
 
 	DescribeTable("container with CPU constraint",
 		func(cpus int, fail bool) {
-			Skip("Issue: https://github.com/kata-containers/tests/issues/250")
 			args = append(args, "--cpus", fmt.Sprintf("%d", cpus), Image, "sh", "-c", "sleep 5; nproc")
 			stdout, _, exitCode := dockerRun(args...)
 			if fail {

--- a/integration/docker/cpu_test.go
+++ b/integration/docker/cpu_test.go
@@ -61,14 +61,22 @@ func withCPUConstraint(cpus float64, defaultVCPUs int, fail bool) TableEntry {
 
 var _ = Describe("Hot plug CPUs", func() {
 	var (
-		args         []string
-		id           string
-		vCPUs        int
-		defaultVCPUs = getDefaultVCPUs()
+		args            []string
+		id              string
+		vCPUs           int
+		defaultVCPUs    = getDefaultVCPUs()
+		cpuSysPath      string
+		waitTime        int
+		maxTries        int
+		checkCpusCmdFmt string
 	)
 
 	BeforeEach(func() {
 		id = RandID(30)
+		cpuSysPath = "/sys/devices/system/cpu"
+		checkCpusCmdFmt = `c=0; while [[ "$(cat %s/cpu%d/online 2> /dev/null)" != "1" ]] && [[ $c < %d ]]; do sleep %d; ((c++)); done; nproc`
+		waitTime = 5
+		maxTries = 5
 		args = []string{"--rm", "--name", id}
 		Expect(defaultVCPUs).To(BeNumerically(">", 0))
 	})
@@ -79,16 +87,17 @@ var _ = Describe("Hot plug CPUs", func() {
 
 	DescribeTable("container with CPU period and quota",
 		func(quota, period int, fail bool) {
+			vCPUs = ((quota + period - 1) / period) + defaultVCPUs
 			args = append(args, "--cpu-quota", fmt.Sprintf("%d", quota),
-				"--cpu-period", fmt.Sprintf("%d", period), Image, "sh", "-c", "sleep 5; nproc")
-			vCPUs = (quota + period - 1) / period
+				"--cpu-period", fmt.Sprintf("%d", period), Image, "sh", "-c",
+				fmt.Sprintf(checkCpusCmdFmt, cpuSysPath, vCPUs-1, maxTries, waitTime))
 			stdout, _, exitCode := dockerRun(args...)
 			if fail {
 				Expect(exitCode).ToNot(BeZero())
 				return
 			}
 			Expect(exitCode).To(BeZero())
-			Expect(fmt.Sprintf("%d", vCPUs+defaultVCPUs)).To(Equal(strings.Trim(stdout, "\n\t ")))
+			Expect(fmt.Sprintf("%d", vCPUs)).To(Equal(strings.Trim(stdout, "\n\t ")))
 		},
 		withCPUPeriodAndQuota(30000, 20000, defaultVCPUs, false),
 		withCPUPeriodAndQuota(30000, 10000, defaultVCPUs, false),
@@ -98,7 +107,8 @@ var _ = Describe("Hot plug CPUs", func() {
 
 	DescribeTable("container with CPU constraint",
 		func(cpus int, fail bool) {
-			args = append(args, "--cpus", fmt.Sprintf("%d", cpus), Image, "sh", "-c", "sleep 5; nproc")
+			args = append(args, "--cpus", fmt.Sprintf("%d", cpus), Image, "sh", "-c",
+				fmt.Sprintf(checkCpusCmdFmt, cpuSysPath, vCPUs-1, maxTries, waitTime))
 			stdout, _, exitCode := dockerRun(args...)
 			if fail {
 				Expect(exitCode).ToNot(BeZero())

